packages/tpetra/core/src/Tpetra_Details_reallocDualViewIfNeeded.hpp

```diff
@@ -100,22 +100,52 @@ reallocDualViewIfNeeded (Kokkos::DualView<ValueType*, DeviceType>& dv,
   }
   else if (curSize < newSize) { // too small; need to reallocate
     if (needFenceBeforeRealloc) {
-      execution_space().fence ();
+      // FENCE REVIEW - NOW PASSING
+      //   Testing: This code is exercised by unit tests.
+      //   GTX960:  Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+      //   White:   Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+      //   Plan:    Remove fence. But not sure yet why fence was placed.
+      //   Notes:
+
+      // execution_space().fence ();
```

Comment:
Since needFenceBeforeRealloc is passed as an input argument, we should respect the intention. Fence is necessary.

Reply:
Kept and added comments #7545